service/matching/physical_task_queue_manager.go

```diff
@@ -466,6 +467,22 @@ func (c *physicalTaskQueueManagerImpl) HasPollerAfter(accessTime time.Time) bool
 	return len(recentPollers) > 0
 }
 
+func (c *physicalTaskQueueManagerImpl) HasVersionedPollerAfter(accessTime time.Time, buildID string) bool {
+	if c.currentPolls.Load() > 0 {
+		return true
+	}
+	if c.pollerHistory == nil {
+		return false
+	}
+	recentPollers := c.pollerHistory.getPollerInfo(accessTime)
+	for _, r := range recentPollers {
+		if r.GetWorkerVersionCapabilities().GetBuildId() == buildID {
```

Comment:
poller might be sending build ID without opting into versioning. We need to check that flag too. if a poller sends UseVersioning=false it's considered unversioned even if it has a build ID.